### PR TITLE
Enhance docs for revision tags and add tests

### PR DESCRIPTION
**Pull request by kfaseela**
Signed-off-by: Faseela K <faseela.k@est.tech>

Enhance docs for revision tags as part of promotion of the feature to Beta

- Enhance the documentation for revision tags
- Add tests for canary upgrades
- Add tests for revision tags

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by ericvn** [comment]
/test doc.test.profile_default

Failed in dashboard

**Comment by ericvn** [comment]
/test doc.test.profile_none

failedin opencensusagent/test.sh/after_snapshot

**Comment by kfaseela** [comment]
> /test doc.test.profile_none
> 
> failedin opencensusagent/test.sh/after_snapshot

This had genuine failures. Just fixed them

**Review by howardjohn — APPROVED** [review]

**Comment by ericvn** [comment]
I haven't looked but is there something in the canary_test (which looks like it is run a few tests before this) leaving things in a not so nice state for the dashboard test, or is it just being flakey? We may not be checking for everything in the before/after logic.

**Comment by kfaseela** [comment]
> I haven't looked but is there something in the canary_test (which looks like it is run a few tests before this) leaving things in a not so nice state for the dashboard test, or is it just being flakey? We may not be checking for everything in the before/after logic.

Thanks! You were right, there were some cleanup issues. Fixed now

File: content/en/boilerplates/revision-tags-usage.md [hunk 1]
```diff
@@ -2,4 +2,4 @@
 ---
 Consider a cluster with two revisions installed, `1-9-5` and `1-10-0`. The cluster operator creates a revision tag `prod-stable`,
 pointed at the older, stable `1-9-5` version, and a revision tag `prod-canary` pointed at the newer `1-10-0` revision. That
-state could be reached via these commands:
\ No newline at end of file
+state could be reached via the below commands:
```
Comment (ericvn) [inline content/en/boilerplates/revision-tags-usage.md:5]: Not sure I like the below commands, maybe the following commands:
```suggestion
state could be reached via the following commands:
```

**Pull request merged**